plugin/pkg/scheduler/algorithm/priorities/spreading.go

```diff
@@ -22,28 +22,42 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/scheduler/algorithm"
 )
 
-type ServiceSpread struct {
-	serviceLister algorithm.ServiceLister
+type JobSpread struct {
+	serviceLister    algorithm.ServiceLister
+	controllerLister algorithm.ControllerLister
 }
 
-func NewServiceSpreadPriority(serviceLister algorithm.ServiceLister) algorithm.PriorityFunction {
-	serviceSpread := &ServiceSpread{
-		serviceLister: serviceLister,
+func NewJobSpreadPriority(serviceLister algorithm.ServiceLister, controllerLister algorithm.ControllerLister) algorithm.PriorityFunction {
+	jobSpread := &JobSpread{
+		serviceLister:    serviceLister,
+		controllerLister: controllerLister,
 	}
-	return serviceSpread.CalculateSpreadPriority
+	return jobSpread.CalculateSpreadPriority
 }
 
-// CalculateSpreadPriority spreads pods by minimizing the number of pods belonging to the same service
-// on the same machine.
-func (s *ServiceSpread) CalculateSpreadPriority(pod *api.Pod, podLister algorithm.PodLister, minionLister algorithm.MinionLister) (algorithm.HostPriorityList, error) {
+// CalculateSpreadPriority spreads pods by minimizing the number of pods belonging to the same service or replication controller if there's no service for
+// given pods on the same machine.
+func (s *JobSpread) CalculateSpreadPriority(pod *api.Pod, podLister algorithm.PodLister, minionLister algorithm.MinionLister) (algorithm.HostPriorityList, error) {
 	var maxCount int
-	var nsServicePods []*api.Pod
+	var nsJobPods []*api.Pod
 
+	var selector labels.Selector
+	doSpreading := false
 	services, err := s.serviceLister.GetPodServices(pod)
 	if err == nil {
 		// just use the first service and get the other pods within the service
 		// TODO: a separate predicate can be created that tries to handle all services for the pod
-		selector := labels.SelectorFromSet(services[0].Spec.Selector)
+		selector = labels.SelectorFromSet(services[0].Spec.Selector)
+		doSpreading = true
+	} else {
```

Comment:
add some comment about what's going on here -- if no service found, then use ReplicationController

also, add comment "just use the first ReplicationController..." just like for the services case